### PR TITLE
fix: issue with dependencies

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

0.5.9 accidentally broke dependencies in `ape-cairo`.
Luckily, caught it fast and fixed it!

### How I did it

Soft implementation of `get_compiler_settings()`.

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: ape_cairo/compiler.py [hunk 1]
```diff
@@ -30,7 +31,19 @@ def name(self) -> str:
 
     @property
     def config(self) -> CairoConfig:
-        return self.config_manager.get_config("cairo")  # type: ignore
+        return cast(CairoConfig, self.config_manager.get_config("cairo"))
+
+    def get_compiler_settings(
```
Comment (unparalleled-js) [inline ape_cairo/compiler.py:36]: @dtdang The real fix was adding this method here so core Ape does not mess up.

File: setup.py [hunk 2]
```diff
@@ -15,6 +15,9 @@
         "types-setuptools",  # Needed due to mypy typeshed
         "flake8>=5.0.4",  # Style linter
         "isort>=5.10.1",  # Import sorting linter
+        "mdformat>=0.7.16",  # Auto-formatter for markdown
```
Comment (unparalleled-js) [inline setup.py:18]: I also added some changes from the project template. You can optionally include these. We are going to have to soon anyways.

File: setup.py [hunk 3]
```diff
@@ -54,10 +57,10 @@
     url="https://github.com/ApeWorX/ape-cairo",
     include_package_data=True,
     install_requires=[
-        "cairo-lang>=0.10.1,<0.11",
-        "starknet.py>=0.10.0a0,<0.11",
-        "eth-ape>=0.5.5,<0.6",
-        "ethpm-types",
+        "cairo-lang>=0.10.3,<0.11",
+        "starknet_py>=0.12.0a0,<0.13",
```
Comment (unparalleled-js) [inline setup.py:61]: In ape-starknet, make sure to bump these if needed to get them to install together correctly

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -2,15 +2,22 @@ on: ["push", "pull_request"]
...
 
         steps:
-        - uses: actions/checkout@v2
+        - uses: actions/checkout@v3
 
         - name: Setup Python
-          uses: actions/setup-python@v2
+          uses: actions/setup-python@v4
           with:
               python-version: 3.8
```
Comment (dtdang) [inline .github/workflows/test.yaml:22]: we have some python-version 3.8 and some 3.9 throughout the files, should we change all to 3.9 to be consistent?

Comment (unparalleled-js) [reply .github/workflows/test.yaml:22]: I think there may be a reason for that, because of commitizen requirements.

**Review by dtdang — APPROVED** [review]

**Pull request merged**